clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1637,6 +1637,27 @@ static std::string eraseAnonNamespace(std::string S) {
   return S;
 }
 
+static bool checkIfDeclInCLNameSpace(DeclContext *DC) {
+
+  bool inCL = false;
+
+  while (DC) {
+    auto *NS = dyn_cast_or_null<NamespaceDecl>(DC);
```

Comment:
What about if this enum is in another container?  It is pretty common to put enums in recorddecl scope, which alters its name as well.

This whole function shouldn't really need to exist (or at least not like this).   See https://clang.llvm.org/doxygen/DeclBase_8cpp_source.html#l00350 for how to implement this.

Reply:
Ok. Based on Prem's and your feedback I think I'll just remove this check.